util/cmake/seqan-config.cmake

```diff
@@ -298,12 +298,9 @@ if ((${CMAKE_SYSTEM_NAME} STREQUAL "Linux") OR (${CMAKE_SYSTEM_NAME} STREQUAL "k
   set (SEQAN_LIBRARIES ${SEQAN_LIBRARIES} rt)
 endif ()
 
+# some OSes don't link pthread fully when building statically so we explicitly include whole archive
 if (UNIX AND NOT APPLE)
-  if ((CMAKE_CXX_FLAGS MATCHES "-static") OR (SEQAN_CXX_FLAGS MATCHES "-static") OR (CMAKE_EXE_LINKER_FLAGS MATCHES "-static"))
     set (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,--whole-archive -lpthread -Wl,--no-whole-archive")
-  else ()
-    set (SEQAN_LIBRARIES ${SEQAN_LIBRARIES} pthread)
-  endif ()
```

Comment:
so we always set whole archive independent of static flag, meaning we always have it statically linked also in `DEVELOP` mode?

Reply:
No, whether or not it is "whole" is independent of static/dynamic. Yes, we are always making it "whole" now, because the way of detecting static is broken, but it doesn't imply that it is always static. In fact when linking dynamically it doesn't make a difference so it only applies to static linking really.

Reply:
ok then